iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/ExecutableNotOwnedByRootCheck.java

```diff
@@ -46,27 +46,40 @@ public class ExecutableNotOwnedByRootCheck implements IacCheck {
   private static final Set<String> SENSITIVE_FILE_EXTENSION = Set.of("exe", "py", "rb", "pl", "lua", "js", "lisp", "sh", "jar", "war",
     "run", "bin", "bat", "ps1");
 
+  private static final Set<String> COMPLIANT_CHOWN_VALUES = Set.of("root", "0", "");
+
   @Override
   public void initialize(InitContext init) {
     init.register(TransferInstruction.class, ExecutableNotOwnedByRootCheck::checkTransferInstruction);
   }
 
   private static void checkTransferInstruction(CheckContext ctx, TransferInstruction transferInstruction) {
     Flag sensitiveChownFlag = getChownFlagSensitive(transferInstruction);
+
     if (sensitiveChownFlag != null) {
       List<Argument> sensitiveFiles = getFilesSensitive(transferInstruction.srcs());
-      Chmod chmod = getChmod(transferInstruction);
 
+      if (isNonRootUserValue(sensitiveChownFlag)) {
+        reportIssue(ctx, sensitiveChownFlag, sensitiveFiles);
+      }
+
+      Chmod chmod = getChmod(transferInstruction);
       if (chmod == null) {
         if (!sensitiveFiles.isEmpty()) {
           reportIssue(ctx, sensitiveChownFlag, sensitiveFiles);
         }
-      } else if (isChmodWriteSensitive(chmod) && (isChmodExecuteSensitive(chmod) || !sensitiveFiles.isEmpty())) {
+      } else if (shouldBeReportedBasedOnChmod(sensitiveChownFlag, sensitiveFiles, chmod)) {
         reportIssue(ctx, sensitiveChownFlag, sensitiveFiles);
       }
     }
   }
 
+  private static boolean shouldBeReportedBasedOnChmod(Flag sensitiveChownFlag, List<Argument> sensitiveFiles, Chmod chmod) {
```

Comment:
```suggestion
  private static boolean isSensitiveChmod(Flag sensitiveChownFlag, List<Argument> sensitiveFiles, Chmod chmod) {
```
IMO `shouldBeReported` is somehow implicit when looking at the condition body.